client/packages/system/src/Immunisation/VaccineCourseEditModal/VaccineCourseEditModal.tsx

```diff
@@ -0,0 +1,289 @@
+
+  const { Modal } = useDialog({ isOpen, onClose, disableBackdrop: true });
+  const height = useKeyboardHeightAdjustment(600);
+
+  const tryUpdateValue = (value: number | undefined) => {
+    if (value === undefined) {
+      return;
+    }
+    updatePatch({ doses: value });
+  };
+
+  // const updateSchedule = (value: number) => {
+  //   if (!value) {
+  //     return;
+  //   }
+  //   const scheduleSeed = (number: number): VaccineCourseScheduleNode => {
+  //     return {
+  //       __typename: 'VaccineCourseScheduleNode',
+  //       id: FnUtils.generateUUID(),
+  //       doseNumber: number,
+  //       label: '',
+  //     };
+  //   };
+  //   let rows = draft?.vaccineCourseSchedules ?? [];
+
+  //   if (rows.length === value) {
+  //     return;
+  //   } else if (value > rows.length) {
+  //     let toAdd = value - rows.length;
+  //     while (toAdd > 0) {
+  //       const number = value - toAdd + 1;
+  //       rows.push(scheduleSeed(number));
+  //       toAdd--;
+  //     }
+  //   } else {
+  //     rows = rows.slice(0, value);
+  //   }
+  //   updatePatch({ vaccineCourseSchedules: rows });
+  // };
+
+  // const updateDescription = (patch: RecordPatch<VaccineCourseScheduleNode>) => {
+  //   if (!patch) {
+  //     return;
+  //   }
+  //   const schedule = {
+  //     ...ArrayUtils.toObject(draft.vaccineCourseSchedules ?? [defaultRow]),
+  //     [patch.id]: patch,
+  //   };
+
+  //   const scheduleArray = Object.values(
+  //     schedule
+  //   ) as VaccineCourseScheduleNode[];
+  //   updatePatch({ vaccineCourseSchedules: scheduleArray });
+  // };
+
+  // const dosesColumns = useColumns<VaccineCourseScheduleNode>(
+  //   [
+  //     { key: 'doseNumber', label: 'label.dose-number' },
+  //     [descriptionColumn(t('label.new-row')), { setter: updateDescription }],
+  //   ],
+  //   {},
+  //   [draft]
+  // );
+
+  const options = useMemo(
+    () => getDemographicOptions(demographicData?.nodes ?? []),
+    [demographicData]
+  );
+
+  if (isLoading) {
+    return <BasicSpinner />;
+  }
+
+  const defaultValue = {
+    value: draft.demographicIndicator?.name ?? '',
+    label: draft.demographicIndicator
+      ? `${draft.demographicIndicator?.name} ${draft.demographicIndicator?.baseYear}`
+      : '',
+  };
+
+  const save = async () => {
+    setIsDirty(false);
+    try {
+      const result =
+        mode === ModalMode.Update
+          ? await update()
+          : await create(programId ?? '');
+      if (result?.__typename === 'VaccineCourseNode') {
+        const message =
+          mode === ModalMode.Update
+            ? `${t('messages.updated-new-vaccine-course')}: ${result.name}`
+            : `${t('messages.created-new-vaccine-course')}: ${result.name}`;
+        success(message)();
+        onClose();
+      }
+    } catch (e) {
+      error(t('error.failed-to-save-vaccine-course'))();
+      console.error(e);
+    }
+  };
+
+  return (
+    <Modal
+      title=""
```

Comment:
I prefer having the title populated and the Section removed
```suggestion
      title={
        mode === ModalMode.Create
          ? t('heading.create-vaccine-course')
          : t('heading.edit-vaccine-course')
      }
```